### ISSUE TITLE
Copying containers failes with error: failed to create mountpoint

### ISSUE THREAD

**Issue by harolds** [lxc/lxd#7758]
When I try to copy an existing container it results in an error. Reading other simular issues did not help me to fix it.

root@dev ~ # lxc copy kasupg5 kasupg3
Error: Create instance from copy: Failed to run: zfs clone tank/containers/kasupg5@copy-47c738fe-5d80-40a7-8d8b-69ec682ab175 tank/containers/kasupg3: cannot mount '/tank/containers/kasupg3': failed to create mountpoint
filesystem successfully created, but not mounted

Same error pops up when I try to launch a fresh container:

root@dev ~ # lxc launch ubuntu:18.04 test
Creating test
Error: Failed instance creation: Create instance from image: Failed to run: zfs clone tank/images/a92eaa65a5c5e53c6bf788b4443f4e5d2afac1665486247c336aa90959522bb6@readonly tank/containers/test: cannot mount '/tank/containers/test': failed to create mountpoint
filesystem successfully created, but not mounted

Some more info:

root@dev ~ # lxc storage list
+------+-------------+--------+--------+---------+
| NAME | DESCRIPTION | DRIVER | SOURCE | USED BY |
+------+-------------+--------+--------+---------+
| tank |             | zfs    | tank   | 20      |
+------+-------------+--------+--------+---------+

root@dev ~ # lxc storage show tank
config:
  source: tank
  zfs.pool_name: tank
description: ""
name: tank
driver: zfs
used_by:
- /1.0/images/a92eaa65a5c5e53c6bf788b4443f4e5d2afac1665486247c336aa90959522bb6
- /1.0/images/e4d11d41a2346917567e55493297b0a37fd96d5a78801f85c4c785493e8dc75a
- /1.0/instances/kasact
- /1.0/instances/kasact2
- /1.0/instances/kasdev
- /1.0/instances/kasdev2
- /1.0/instances/kasdev3
- /1.0/instances/kasdev4
- /1.0/instances/kasdev5
- /1.0/instances/kasdev6
- /1.0/instances/kasdev7
- /1.0/instances/kasdev8
- /1.0/instances/kasedsntest
- /1.0/instances/kastest
- /1.0/instances/kastest2
- /1.0/instances/kasupg2
- /1.0/instances/kasupg4
- /1.0/instances/kasupg5
- /1.0/instances/kasupg7
- /1.0/profiles/default
status: Created
locations:
- none

**Comment by stgraber** [comment]
This error normally indicates that you have altered the `mountpoint` property of some of your containers, in this case it's showing `/tank`.

LXD only supports having datasets mounted where it sets them to be `/var/snap/lxd/common/lxd/storage-pools/POOL/...`.

Showing `zfs list -t all -o name,mountpoint` should make it quite visible where things have gone wrong.

**Comment by harolds** [comment]
Thanks that helped! It wasn't the mountpoint for the containers by the 'main' mountpoints.
For unknow reasons their where wrong. As far as I can remember I did not change it manually.

root@dev ~ # zfs list -t all -o name,mountpoint
NAME                                                                                           MOUNTPOINT
tank                                                                                           /tank
tank/containers                                                                                /tank/containers

Fixed it by using:
root@dev ~ # zfs set mountpoint=none tank
root@dev ~ # zfs set mountpoint=none tank/containers

Result:
root@dev ~ # zfs list -t all -o name,mountpoint
NAME                                                                                           MOUNTPOINT
tank                                                                                           none
tank/containers                                                                                none
